pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -846,16 +846,22 @@ protected boolean commitSegment(String controllerVipUrl, boolean isSplitCommit)
     if (isSplitCommit) {
       // TODO: make segment uploader used in the segment committer configurable.
       SegmentUploader segmentUploader;
-      try {
-        segmentUploader =
-            new Server2ControllerSegmentUploader(segmentLogger, _protocolHandler.getFileUploadDownloadClient(),
-                _protocolHandler.getSegmentCommitUploadURL(params, controllerVipUrl), _segmentNameStr,
-                ServerSegmentCompletionProtocolHandler.getSegmentUploadRequestTimeoutMs(), _serverMetrics);
-      } catch (URISyntaxException e) {
-        segmentLogger.error("Segment commit upload url error: ", e);
-        return SegmentCompletionProtocol.RESP_NOT_SENT;
+      if (this._tableConfig.getValidationConfig().getPeerSegmentDownloadScheme() != null) {
```

Comment:
Move this logic into the SegmentCommitterFactory class. Instantiate the class with `tableConfig`
Change the factory to have a single call to create committer, with `params` and `isSplit`. The uploader can be constructed inside the factory (in the split case).

Reply:
Done. thanks for the suggestion.. Much cleaner codes now.